src/main/java/org/hydev/mcpm/client/Downloader.java

```diff
@@ -38,10 +84,25 @@ public void downloadFile(String url, File to)
      * @param progress Show progress or not
      * @param threads Number of simultaneous downloads
      */
-    public void downloadFiles(Map<String, File> urls, boolean progress, int threads)
-    {
-        // TODO: Implement this
-        throw new UnsupportedOperationException("TODO");
+    public void downloadFiles(Map<String, File> urls, boolean progress, int threads) throws IOException {
+        ExecutorService executor = Executors.newFixedThreadPool(threads);
+        var files = urls.keySet().stream().toList();
+        if (files.size() > 0) {
+            for (int i=0; i<files.size(); i++) {
+                executor.submit(new Processor(i, urls, files));
+            }
+            executor.shutdown();
+            // All files submitted for downloading
+
+
+            try{
+                executor.awaitTermination(20, TimeUnit.SECONDS);
```

Comment:
Does this mean we'll count files that take longer than 20 seconds to download as a failure?

Reply:
Oh, it meant we wanted to wait for the downloading file complete in the thread for the 20s before executing the following body method can execute.

Reply:
What if the thread takes longer than 20 seconds?

Reply:
Then it's doom. But I saw there is a suggestion to use the following to make sure threads finish.

while (!executor.isTerminated()) {}